src/Chainweb/Payload/PayloadStore.hs

```diff
@@ -284,14 +284,7 @@ instance (pk ~ CasKeyType (PayloadWithOutputs_ a), CanReadablePayloadCas_ a tbl)
         outs <- MaybeT $ tableLookup
             (_payloadCacheBlockOutputs $ _payloadCache db)
             outsHash
-        return $ PayloadWithOutputs
-            { _payloadWithOutputsTransactions = V.zip (_blockTransactions txs) (_blockOutputs outs)
-            , _payloadWithOutputsMiner = _blockMinerData txs
-            , _payloadWithOutputsCoinbase = _blockCoinbaseOutput outs
-            , _payloadWithOutputsPayloadHash = k
-            , _payloadWithOutputsTransactionsHash = txsHash
-            , _payloadWithOutputsOutputsHash = outsHash
-            }
+        return $ newPayloadWithOutputs (_blockMinerData txs) (_blockCoinbaseOutput outs) (V.zip (_blockTransactions txs) (_blockOutputs outs))
```

Comment:
So we're recomputing the hashes now? Can we avoid that?

Reply:
Yep, by using the neighbour smart constructor `payloadWithOutputs`.